package.json

```diff
@@ -36,6 +36,7 @@
     "ioredis": "^4.27.6",
     "js-file-download": "^0.4.11",
     "js-yaml": "^4.1.0",
+    "json2csv": "^3.4.2",
```

Comment:
This dependency has been abandoned. https://www.npmjs.com/package/json2csv

Can we write this ourselves?

Reply:
We already had this dependency through the similarly-named `json-to-csv`, which is required on the next line. I added `json2csv` here to make it explicit, because I am requiring it directly. `yarn.lock` didn't have to change with this change to `package.json`.

The earlier dependency, `json-to-csv`, is used by `scripts/exportSitesAsCsv.js`. The reason I didn't just use the same approach is that the `json-to-csv` library writes to an output file, whereas for this use case we needed the CSV content as a string to serve through the CSV. The underlying `json2csv` library provided this already to `json-to-csv` so I went ahead and used it.

We can certainly render our own CSV in either or both of our existing use cases, but we have to write our own code and suitable tests for the stuff (quoting, escaping, headers, missing fields) that's already implemented. It's a tradeoff.

Reply:
Honestly, the `json-to-csv` package we were already directly depending to write an output file consists of very little code and could be easily replaced.

But the `json2csv` library [has a lot more going on](https://www.npmjs.com/package/json2csv?activeTab=code) with formatters for different field types and other logic. We don't need most of that in each of our specific uses but if we're going to be adding more reports in the future I think sooner or later we'll end up re-solving a lot of details that a decent library will have already addressed.

Potential approach: we could move to an actively maintained library ([example](https://www.npmjs.com/package/json-2-csv)) for the JSON/CSV logic, and write our own code for the file output stage that's needed by `scripts/exportSitesAsCsv.js`. We just have to decide whether we do part or all of that on this PR.

Reply:
@apburnes As of 5c0f55d this PR is dependent on the current (v7.0.1) version of the maintained, GitHub-hosted version of `json2csv`. 

The `exportSitesAsCsv.js` script is still indirectly dependent on the legacy version. I'll be glad to tackle fixing that too but I don't think it belongs on this PR. If it's all right with you I'll make a separate issue for that.

Reply:
I see. Can you update the dependency to the new package names with `@json2csv/transforms` and `@json2csv/plainjs` like the example in https://www.npmjs.com/package/@json2csv/transforms?

Reply:
I'm not clear on what you're asking for, @apburnes. I'm not using any of the transforms so I'm only requiring the `plainjs` package. If you mean you'd like me to use the import syntax instead of require, I'm afraid I tried that and got `SyntaxError: Cannot use import statement outside a module` which @drewbo concluded was probably because we don’t bundle our API to support imports.

Reply:
If we just need `@json2csv/plainjs` functionality then add that dependency and require it by

```js
const json2csv = require('@json2csv/plainjs');
const parser = new json2csv.Parser(opts);
const csv = parser.parse(data);
```

Reply:
I'm still a little confused because I got that 
```js
{ Parser } from '@json2csv/plainjs'
```
syntax from the docs you linked to.

But I have now changed (d617c4d) the latest code here from 
```js
const { Parser } = require('@json2csv/plainjs');
const parser = new Parser({ fields });
const csv = parser.parse(domains);
```
to
```js
const json2csv = require('@json2csv/plainjs');
const parser = new json2csv.Parser({ fields });
const csv = parser.parse(domains);
```
and it continues to function.

Reply:
`const { Parser } from '@json2csv/plainjs'` is es module syntax. The app currently only supports commonjs syntax like `const { Parser } = require('@json2csv/plainjs');`.